### DISCUSSION TITLE
Options for the Download and Generation of the Video, Audio, TXT, PDF and Midi files.

### DISCUSSION THREAD

**Discussion by mARTin-B78** [rakuri255/UltraSinger#177] [Ideas]
Currently there are these options

    [opt]
    -h      This help text.
    -i      Ultrastar.txt
            audio like .mp3, .wav, youtube link
            youtube Playlist Link # Feature Request
            youtube Playlist txt file # Feature Request
    -o      Output folder

    [mode]
    ## if INPUT is audio ##
    default  Creates all

    # Single file creation selection is in progress, you currently getting all!
    (-u      Create ultrastar txt file) # In Progress
    (-m      Create midi file) # In Progress
    (-s      Create PDF sheet file) # In Progress
    

    It would be nice to have some more in the Future
    (-dv     Download Video file) # Feature Request
    (-da      Download Audio file) # Feature Request
    (-dc     Download Coverart) # Feature Request
    (-ds     Download Subtitle Tracks) # Feature Request
    (-dl     Save link to Video Source) # Feature Request    
    (-ci      Create, demux instrumental Audio file) # Feature Request
    (-cv      Create, demux vocal audio file)# Feature Request

What do you guys think?

**Comment by rakuri255** [comment]
dl, ci, cv ist implemented if you use UltraStar Format Version 1.1.0 ```--format_version 1.1.0```

For the others, can you explain the Feature Request a bit more?

**Comment by mARTin-B78** [comment]
Right now, the manual work required to correct lyrics after speech recognition is still pretty high. 
Imagine if UltraSinger could streamline the entire process, from obtaining the lyrics to producing the final karaoke-ready file – all in a single batch workflow! 

Some lyrics can already be sourced from databases like USDB  https://usdb.animux.de/ or other Databases.
But once you have the lyrics, other components like the video (with or without sound), audio tracks (with vocals, without vocals, or instrumental), and cover art or even a midi file or a PDF sheet music file are still needed. 
UltraSinger could bring all these elements together automatically!

Suggested Features:

-u Generate the TXT file (or skip if one already exists, or update an existing one)
-m Create a MIDI file, if desired
-s Generate a PDF sheet music file if desired

UltraSinger could then automatically download or extract any missing components directly from YouTube:
-dv Download video from a YouTube playlist or a text file with video links
-da Download or extract the audio track (from the playlist, text file, or directly from the video)
-dc Download cover art (already implemented)
-ds Download subtitle tracks to help match or correct lyrics when they differ from the actual video text
-dl Save the video source link, either as a link file in the folder or as a comment in the TXT file

With these features, UltraSinger could become the ultimate tool for karaoke enthusiasts and DIY musicians alike!
